napari/layers/labels/labels.py

```diff
@@ -1068,14 +1072,14 @@ def _raw_to_displayed(
         if labels_to_map.size == 0:
             return self._cached_mapped_labels[data_slice]
 
-        if self.color_mode == LabelColorMode.AUTO:
-            mapped_labels = _cast_labels_data_to_texture_dtype_auto(
-                labels_to_map, self._random_colormap
-            )
+        if new_label is not None and self.contour < 1:
+            if new_label not in self._cached_labels_mapping:
+                map_ = self._cast_labels_using_colormap(np.array([new_label]))
+                self._cached_labels_mapping[new_label] = map_[0], map_.dtype
+            val = self._cached_labels_mapping[new_label]
+            mapped_labels = np.full_like(labels_to_map, val[0], dtype=val[1])
```

Comment:
As in other PRs, I don't think you need to wrap the new_label in an array only to grab the 0th element? ie this should work:

```suggestion
                map_ = self._cast_labels_using_colormap(new_label)
                self._cached_labels_mapping[new_label] = map_, map_.dtype
            val, dt = self._cached_labels_mapping[new_label]
            mapped_labels = np.full_like(labels_to_map, val, dtype=dt)
```

Reply:
Oh, typing. Guh. I'll leave you to deal with that. 😂

Reply:
Not only typing. Pure python int does not have `dtype` attribute.